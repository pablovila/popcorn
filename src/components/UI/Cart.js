import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faShoppingCart } from "@fortawesome/free-solid-svg-icons";

const cart = () => {
  return (
    <>
      <button className="button is-info">
        <FontAwesomeIcon icon={faShoppingCart} />
      </button>
    </>
  );
};

export default cart;
