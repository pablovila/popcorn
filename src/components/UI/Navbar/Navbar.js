import React from "react";
import PropTypes from "prop-types";

const navbar = ({ children }) => {
  return (
    <nav className="navbar" role="navigation" aria-label="main navigation">
      <div className="container">{children}</div>
    </nav>
  );
};

navbar.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ]).isRequired
};

export default navbar;
