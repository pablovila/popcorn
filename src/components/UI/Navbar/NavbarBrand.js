import React from "react";
import PropTypes from "prop-types";

const navbarBrand = ({ children }) => {
  return <div className="navbar-brand">{children}</div>;
};

navbarBrand.propTypes = {
  children: PropTypes.element
};

export default navbarBrand;
