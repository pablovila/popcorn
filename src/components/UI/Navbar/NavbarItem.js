import React from "react";
import PropTypes from "prop-types";

const navbarItem = props => {
  const { children, ...aProps } = props;
  return (
    <div className="navbar-item" {...aProps}>
      {children}
    </div>
  );
};

navbarItem.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired
};

export default navbarItem;
