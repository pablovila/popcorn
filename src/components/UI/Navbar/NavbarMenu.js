import React from "react";
import PropTypes from "prop-types";

const navbarMenu = ({ children }) => {
  return (
    <div className="navbar-menu">
      <div className="navbar-end">{children}</div>
    </div>
  );
};

navbarMenu.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ])
};

export default navbarMenu;
