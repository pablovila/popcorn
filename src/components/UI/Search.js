import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

const search = () => {
  return (
    <div className="field has-addons">
      <div className="control has-icons-left">
        <input className="input" type="text" placeholder="Find a movie" />
        <span className="icon is-left">
          <FontAwesomeIcon icon={faSearch} />
        </span>
      </div>
      <div className="control">
        <button className="button is-info">Search</button>
      </div>
    </div>
  );
};

export default search;
