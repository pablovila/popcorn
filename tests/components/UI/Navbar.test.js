import React from "react";
import { mount } from "enzyme";

import Navbar from "components/UI/Navbar/Navbar";
import NavbarBrand from "components/UI/Navbar/NavbarBrand";
import NavbarItem from "components/UI/Navbar/NavbarItem";
import NavbarMenu from "components/UI/Navbar/NavbarMenu";

import Search from "components/UI/Search";
import Cart from "components/UI/Cart";

describe("test Navbar component", () => {
  it("renders Navbar component", () => {
    const navbar = mount(
      <Navbar>
        <NavbarBrand />
        <NavbarMenu />
      </Navbar>
    );

    expect(navbar.find(".navbar")).toHaveLength(1);
    expect(navbar).toMatchSnapshot();
  });

  it("renders Navbar children and UI subcomponents", () => {
    const navbar = mount(
      <Navbar>
        <NavbarBrand>
          <NavbarItem>Popcorn Movies</NavbarItem>
        </NavbarBrand>
        <NavbarMenu>
          <NavbarItem>
            <Search />
          </NavbarItem>
          <NavbarItem>
            <Cart />
          </NavbarItem>
        </NavbarMenu>
      </Navbar>
    );
    expect(navbar.find(".navbar")).toHaveLength(1);
    expect(navbar.find(".navbar-brand")).toHaveLength(1);
    expect(navbar.find(".navbar-menu")).toHaveLength(1);
    expect(navbar.find(".navbar-item")).toHaveLength(3);
    expect(navbar.find(".navbar-item input").html()).toEqual(
      '<input class="input" type="text" placeholder="Find a movie">'
    );
  });
});
