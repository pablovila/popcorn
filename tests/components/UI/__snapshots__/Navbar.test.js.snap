// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navbar component renders Navbar component 1`] = `ReactWrapper {}`;
